### PR TITLE
#187 Add possibility to configure the standard download folder

### PR DESCRIPTION
**Pull request by oomelianchuk**
This PR contains changes to enable browser preferences configuration for chrome and firefox and therefore changing the default download directory for these browsers.

### CODE REVIEW

**Review by occupant23 — CHANGES_REQUESTED** [review]
Please divide the changes into two different PRs as discussed.
One only setting the downloads folder just via one configuration.
One introducing the a general approach to set preferences. (Create a separate ticket for this)

**Comment by oomelianchuk** [comment]
Support of browser preferences has been removed from this PR and moved to PR #193

File: src/main/java/com/xceptance/neodymium/module/statement/browser/multibrowser/BrowserRunnerHelper.java [hunk 1]
```diff
@@ -229,6 +236,15 @@ else if (Neodymium.configuration().useProxy())
                     options.addArguments(config.getArguments());
                 }
 
+                if (config.getDownloadDirectory() != null && !config.getDownloadDirectory().isEmpty())
```
[the remark this answers is not in the document]
Comment (oomelianchuk) [reply src/main/java/com/xceptance/neodymium/module/statement/browser/multibrowser/BrowserRunnerHelper.java]: no, used now, thanks for the hint :)

File: src/main/java/com/xceptance/neodymium/module/statement/browser/multibrowser/BrowserRunnerHelper.java [hunk 2]
```diff
@@ -241,7 +257,16 @@ else if (firefoxBrowsers.contains(browserName))
                 {
                     options.addArguments(config.getArguments());
                 }
+                if (config.getDownloadDirectory() != null && !config.getDownloadDirectory().isEmpty())
```
[the remark this answers is not in the document]
Comment (oomelianchuk) [reply src/main/java/com/xceptance/neodymium/module/statement/browser/multibrowser/BrowserRunnerHelper.java]: no, used now, thanks for the hint :)

File: src/main/java/com/xceptance/neodymium/module/statement/browser/multibrowser/BrowserRunnerHelper.java [hunk 3]
```diff
@@ -372,4 +396,19 @@ public static Proxy createProxyCapabilities()
         webdriverProxy.setNoProxy(Neodymium.configuration().getProxyBypass());
         return webdriverProxy;
     }
+
+    private static String popularContentTypes()
```
[the remark this answers is not in the document]
Comment (oomelianchuk) [reply src/main/java/com/xceptance/neodymium/module/statement/browser/multibrowser/BrowserRunnerHelper.java:407]: done

File: src/main/java/com/xceptance/neodymium/module/statement/browser/multibrowser/configuration/BrowserConfigurationMapper.java [hunk 4]
```diff
@@ -330,7 +335,14 @@ else if (!StringUtils.isEmpty(globalHeadless))
             browserConfiguration.setArguments(args);
         }
 
-        capabilities.setCapability("name", browserProfileConfiguration.get("name"));
```
[the remark this answers is not in the document]
Comment (oomelianchuk) [reply src/main/java/com/xceptance/neodymium/module/statement/browser/multibrowser/configuration/BrowserConfigurationMapper.java:333]: hm, I also don't see any reason why the line should be deleted. The change is reverted

File: src/test/java/com/xceptance/neodymium/testclasses/webDriver/DownloadFilesInDifferentWays.java [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+    {
+        fileName = new File("target/02_2020-Java_aktuell-Autor-Rene_Schwietzke-High-Performance-Java-Hinter-den-Kulissen-von-Java.pdf");
+        Selenide.open("https://blog.xceptance.com/2020/02/28/ijug-magazin-java-aktuell-high-performance-java/");
+        $(".alignright.is-resized").scrollIntoView(true).click();
+        waitForFileDownloading();
+        validateFilePresentInDownloadHistory();
+    }
+
+    @Test
+    public void downloadOnFormSubmission()
```
[the remark this answers is not in the document]
Comment (oomelianchuk) [reply src/test/java/com/xceptance/neodymium/testclasses/webDriver/DownloadFilesInDifferentWays.java:52]: Done

File: src/test/java/com/xceptance/neodymium/testclasses/webDriver/DownloadFilesInDifferentWays.java [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+import com.codeborne.selenide.Selenide;
+import com.xceptance.neodymium.NeodymiumRunner;
+import com.xceptance.neodymium.module.statement.browser.multibrowser.Browser;
+import com.xceptance.neodymium.tests.NeodymiumTest;
+import com.xceptance.neodymium.util.Neodymium;
+
+@RunWith(NeodymiumRunner.class)
+@Browser("chrome_download")
+@Browser("firefox_download")
+public class DownloadFilesInDifferentWays extends NeodymiumTest
```
[the remark this answers is not in the document]
Comment (oomelianchuk) [reply src/test/java/com/xceptance/neodymium/testclasses/webDriver/DownloadFilesInDifferentWays.java:31]: JavaDoc is added

File: src/test/java/com/xceptance/neodymium/tests/DownloadFilesExecutorTest.java [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.JUnitCore;
+import org.junit.runner.Result;
+
+import com.xceptance.neodymium.module.statement.browser.multibrowser.configuration.MultibrowserConfiguration;
+import com.xceptance.neodymium.testclasses.webDriver.DownloadFilesInDifferentWays;
+
+public class DownloadFilesExecutorTest extends NeodymiumTest
```
[the remark this answers is not in the document]
Comment (oomelianchuk) [reply src/test/java/com/xceptance/neodymium/tests/DownloadFilesExecutorTest.java:20]: JavaDoc is added

**Review by wurzelkuchen — APPROVED** [review]
looks good, thanks

**Pull request merged**